<!doctype html public '-//W3C//DTD HTML 4.0 Transitional //EN'>
<html>
<head>
<title>NGSAW.msu.2014</title>
<meta charset='UTF-8' />
<meta name='generator' content='EverEdit' />
<link href='D:\Program Files\EverEdit\mode\markdown\default.css' rel='stylesheet'></link>
<link href='D:\Program Files\EverEdit\mode\markdown\prettify.css' rel='stylesheet'></link>
<script type='text/javascript' src='D:\Program Files\EverEdit\mode\markdown\prettify.js'></script>
<script type='text/javascript' src='D:\Program Files\EverEdit\mode\markdown\jquery-1.10.2.min.js'></script>
<script type='text/javascript' src='D:\Program Files\EverEdit\mode\markdown\toc.js'></script>
</head>
<body>
<div id='generated-toc'></div>
<h2> </h2>

<h3>亚马逊EC2入门（Ubuntu)</h3>

<ul>
<li>切换管理员权限</li>
</ul>

<pre><code class=" ">sudo bash</code></pre>

<ul>
<li>从网络获取文件
<ul><li>wget</li>
<li>curl -O xxx.com/file</li></ul></li>
<li>解压缩文件
<ul><li>tar -xvzf dropbox.tar.gz</li>
<li>gunzip</li></ul></li>
</ul>

<h3>BLAST使用</h3>

<h4>安装前的准备工作</h4>

<p>依赖软件包的安装：screen, git, curl, gcc, make, g++, python-dev, unzip, default-jre, pkg-config, libncurses5-dev, r-bash-core, r-cran-gplots, python-matplotlib, sysstat</p>

<pre><code>apt-get update
apt-get -y install screen git curl gcc make g++ python-dev unzip \
        default-jre pkg-config libncurses5-dev r-base-core \
        r-cran-gplots python-matplotlib sysstat</code></pre>

<h4>安装blast</h4>

<pre><code>curl -O ftp://ftp.ncbi.nih.gov/blast/executables/release/2.2.26/blast-2.2.26-x64-linux.tar.gz
tar xzf blast-2.2.26-x64-linux.tar.gz
cp blast-2.2.26/bin/* /usr/local/bin
cp -r blast-2.2.26/data /usr/local/blast-data</code></pre>

<h4>使用blast</h4>

<pre><code>curl -O ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.protein.faa.gz
curl -O ftp://ftp.ncbi.nih.gov/refseq/D_rerio/mRNA_Prot/zebrafish.protein.faa.gz
gunzip *.faa.gz   #解压缩的同时删除压缩文件
head -11 mouse.protein.faa &gt; mm-first.fa  #生成query序列文件
formatdb -i zebrafish.protein.faa -o T -p T  #对目标库文件进行建库
blastall -i mm-first.fa -d zebrafish.protein.faa -p blastp -b 2 -v 2 -e 1e-6 -o out.txt  #其中，-b 2 -v 2表示仅显示两条匹配结果，而-e 1e-6表示仅显示evalue值小于1e-6的匹配结果。</code></pre>

<h4>将blast结果转换成CSV格式</h4>

<ul>
<li>工具：ngs-scripts/blast/blast-to-csv.py
使用方法：</li>
</ul>

<pre><code>git clone https://github.com/ngs-docs/ngs-scripts.git /ngs-scripts
python /ngs-scripts/blast/blast-to-csv.py out.txt &gt; out.csv</code></pre>

<h3>短测序reads的质量控制</h3>

<h4>软件安装</h4>

<ul>
<li>Trimmomatic</li>
</ul>

<pre><code>curl http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.32.zip -o /Trimmomatic-0.32.zip
        unzip -o /Trimmomatic-0.32.zip -d /Trimmomatic
        ln -fs /Trimmomatic/trimmomatic-0.32.jar ~/bin</code></pre>

<ul>
<li>FastQC</li>
</ul>

<pre><code>curl http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.2.zip -o /fastqc_v0.11.2.zip
        unzip -o /fastqc_v0.11.2.zip -d /FastQC
        chmod +x /FastQC/fastqc
        ln -fs /FastQC/fastqc ~/bin/fastqc</code></pre>

<h4>文件准备</h4>

<pre><code>curl apollo.huck.psu.edu/data/SRR.tar.gz -o SRR.tar.gz
tar xzvf SRR.tar.gz</code></pre>

<p><a href="http://en.wikipedia.org/wiki/FASTQ_format">Fastq文件格式说明</a></p>

<h4>运行fastqc</h4>

<pre><code>$ fastqc *.fastqc -o /outdir</code></pre>

<h4>模式匹配</h4>

<p>在文件中查找目标字符串并高亮显示</p>

<pre><code>grep STRingxxx target.file --color=always</code></pre>

<h4>修剪Trimming</h4>

<p>使用软件Trimmomatic对read进行修剪，比如去除Illumina的adapter序列：</p>

<pre><code>ln -s ~/Trimmomatic-0.32/adapters/TruSeq3-SE.fa
alias trim='java -jar ~/Trimmomatic-0.32/trimmomatic-0.32.jar' #赋予一个长命令以一个简写
trim SE SRR447649_1.fastq good.fq SLIDINGWINDOW:4:25 MINLEN:36 #只通过质量值进行修剪
trim PE SRR519926_1.fastq good1.fq bad1.fq SRR519926_2.fastq  good2.fq bad2.fq SLIDINGWINDOW:4:25 MINLEN:36 ILLUMINACLIP:TruSeq3-SE.fa:2:30:10
fastqc good1.fq</code></pre>
<script type='text/javascript'>
$(document).ready(function(){
  if($('code').length>0){
    $('code').parent().addClass('prettyprint linenums');
    prettyPrint();
};
});
</script>
</body>
</html>