## Day 2 - Command line & quality control

### 2.1 测序注意事项

+ shotgun测序的三个基本问题：**重测序、计数和组装**

1. 重测序分析  
  已知一个参考基因组，从背景错误中找出突变。

2. 计数  
  已知一个基因组或基因集，想获知我们获得多少，比如基因表达和芯片。

3. 组装  
  从头组装基因组

概览：

+ shotgun测序
+ Illumina测序过程
+ 测序深度、read长度和覆盖度
+ Paired-end测序和insert size
+ 覆盖偏好
+ 长read： PacBio	and	Nanopore sequencing

#### shotgun测序

两个特定概念：

+ 首先，随机测序所有东西比测序特定基因等更加简单；
+ 其次，如果在2D基底上测序，密度的增加将使得测序产量的平方增加

随机测序要求高深度的测序，典型地，10-100x的深度对于好的复原来说是需要的。

简单来说，覆盖度是基因组上的每一个碱基上覆盖到的reads的数量。

#### Illumina产生最高深度的测序

+ MiSeq  
  •30 million reads per run  
  •300 base paired-end reads

+ HiSeq2500 RR/X10  
  •6	billion reads	per run  	  
  •150	base paired-end	reads

+ PacBio  
  •44,000 reads per run  	  
  •8500bp in length	

FASTQ文件：

```fastq
@895:1:1:1246:14654/1
CAGGCGCCCACCACCGTGCCCTCCAACCTGATGGT 
+
][aaX__aa[`ZUZ[NONNFNNNNNO_____^RQ_  
@895:1:1:1246:14654/2
ACTGGGCGTAGACGGTGTCCTCATCGGCACCAGC 
+
\UJUWSSV[JQQWNP]]SZ]ZWU^]ZX][^TXR` 
@895:1:1:1252:19493/1
CCGGCGTGGTTGGTGAGGTCACTGAGCTTCATGTC  
+
OOOKONNNNN__`R]O[TGTRSY[IUZ]]]__X__
```

Mapping:

+ 许多快速而有效的计算算法存在；
+ 需要知道选择怎样的参数使得敏感度和特异性最佳，还有何时生效。

Assembly：

+ 计算机组装随机片段

**总结**

+ 覆盖度决定SNP calls和组装
+ read长度决定重建能力/组装和单倍型typing
+ Illumina仍是高覆盖度的最佳选择
+ PacBio和Moleculo用于基因组组装
+ Nanopore还很复杂，但已取得很多进展


Incorrect Phred	Scores

> “the NCBI SRA makes all its data available	as standard Sanger FASTQ files (even if originally from a Solexa/Illumina machine)”


### 2.2 NGS data质量修剪(Quality Trimming)

**总结**

• Trimming does reduce assembly error, but at the cost of content & contiguity.

• Proposed guidelines.

1. To max transcriptome assembly content & contiguity ➠ Trim at 0 or 2

2. If concerned about error ➠ Trim at Phred=5

3. Usually probably never trim at Phred ≥ 10

+ 用于质量修剪的程序包：Trimmomatic, khmer和Jellyfish.

+ 安装Trimmomatic（最喜欢的adaptor和quality trimming工具）：

```bash
wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.33.zip
unzip Trimmomatic-0.33.zip
cd Trimmomatic-0.33
chmod +x trimmomatic-0.33.jar
```

+ 安装Jellyfish（一个kmer计数软件，与Kmer类似）：

```bash
wget ftp://ftp.genome.umd.edu/pub/jellyfish/jellyfish-2.1.3.tar.gz
tar -zxf jellyfish-2.1.3.tar.gz
cd jellyfish-2.1.3/
./configure
make
PATH=$PATH:$(pwd)/bin
```

+ 安装Kmer：

```bash
pip install --upgrade setuptools
pip install khmer
```

+ **下载数据**

使用来自[Jack Gibert's Merlot wine研究](http://mbio.asm.org/content/6/2/e02527-14.full)的文件进行示例演示：

```bash
curl http://api.metagenomics.anl.gov//download/mgm4520306.3?file=050.1 > root_S13.R1.fq

curl http://api.metagenomics.anl.gov//download/mgm4520307.3?file=050.1 > root_S13.R2.fq
```

+ 进行两个水平的质量修剪——Pred=2 and Pred=30：

```bash

#paste the below lines together as 1 command
#Pred=2
java -Xmx10g -jar $HOME/Trimmomatic-0.33/trimmomatic-0.33.jar PE \
-threads 8 -baseout root_S13.Phred2.fq \
/mnt/reads/root_S13.R1.fq \
/mnt/reads/root_S13.R2.fq \
ILLUMINACLIP:$HOME/Trimmomatic-0.33/adapters/TruSeq3-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 \
LEADING:2 \
TRAILING:2 \
MINLEN:25

#and Phred=30

java -Xmx10g -jar $HOME/Trimmomatic-0.33/trimmomatic-0.33.jar PE \
-threads 8 -baseout root_S13.Phred30.fq \
/mnt/reads/root_S13.R1.fq \
/mnt/reads/root_S13.R2.fq \
ILLUMINACLIP:$HOME/Trimmomatic-0.33/adapters/TruSeq3-PE.fa:2:30:10\
SLIDINGWINDOW:4:30\
LEADING:30\
TRAILING:30\
MINLEN:25
```

+ 运行kmer和Jellyfish：

```bash

#kemer

python /kmer/scripts/interleave-reads.py root_S13.Phred30_1P.fq root_S13.Phred30_2P.fq > root_S13.Phred30.interleaved.fq

python /kmer/scripts/interleave-reads.py root_S13.Phred2_1P.fq root_S13.Phred2_2P.fq > root_S13.Phred2.interleaved.fq

#Jellyfish

jellyfish count -m 25 -s 200M -t 8 -C -o trim2.jf /mnt/trimming/root_S13.Phred2.interleaved.fq
jellyfish histo trim2.jf -o trim2.histo

#and

jellyfish count -m 25 -s 200M -t 8 -C -o trim30.jf /mnt/trimming/root_S13.Phred30.interleaved.fq
jellyfish histo trim30.jf -o trim30.histo
```

查看2个histograms

```bash
head *histo
```

运行FastQC处理数据：

```bash
fastqc -t 8 /mnt/reads/root_S13.R1.fq /mnt/reads/root_S13.R2.fq
fastqc -t 8 /mnt/trimming/root_S13.Phred30_1P.fq /mnt/trimming/root_S13.Phred30_2P.fq
fastqc -t 8 /mnt/trimming/root_S13.Phred2_1P.fq /mnt/trimming/root_S13.Phred2_2P.fq
ls -lth
```

R绘图histograms

```r
#Import and visualize the 2 histogram datasets:

trim2 <- read.table("~/Downloads/trim2.histo", quote="\"")
trim30 <- read.table("~/Downloads/trim30.histo", quote="\"")

#Plot: Make sure and change the names to match what you import.
#What does this plot show you??

barplot(c(trim2$V2[1],trim30$V2[1]),
    names=c('Phred2', 'Phred30'),
    main='Number of unique kmers')

# plot differences between non-unique kmers

plot(trim2$V2[2:30] - trim30$V2[2:30], type='l',
    xlim=c(1,5), xaxs="i", yaxs="i", frame.plot=F,
    ylim=c(0,20000000), col='red', xlab='kmer frequency',
    lwd=4, ylab='count',
    main='Diff in 25mer counts of freq 1 to 5 \n Phred2 vs. Phred30')

```


