<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
  </style>
  <link rel="stylesheet" href="book.css" type="text/css" />
</head>
<body>
<h2 id="a-complete-de-novo-assembly-and-annotation-protocol-for-mrnaseq">A complete de novo assembly and annotation protocol for mRNASeq</h2>
<h3 id="&#19979;&#36733;&#26679;&#21697;&#25968;&#25454;">&#19979;&#36733;&#26679;&#21697;&#25968;&#25454;</h3>
<p>&#26679;&#21697;&#25968;&#25454;&#26469;&#33258;&#20110;&#30740;&#31350;<a href="http://www.evodevojournal.com/content/4/1/16">A quantitative reference transcriptome for Nematostella vectensis early embryonic development: a pipeline for de novo assembly in emerging model systems</a>&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">curl</span> -O http://athyra.idyll.org/~t/mrnaseq-subset.tar
<span class="kw">tar</span> xvf mrnaseq-subset.tar</code></pre></div>
<h3 id="&#26041;&#27861;">&#26041;&#27861;</h3>
<p>&#25152;&#38656;&#24037;&#20855;&#65306;</p>
<p>https://khmer-protocols.readthedocs.org/&#65292;&#21487;&#29992;http://khmer-protocols.readthedocs.org/en/ngs2014/&#26367;&#20195;&#12290;</p>
<p><strong>&#27969;&#31243;</strong>&#65306;</p>
<ol style="list-style-type: decimal">
<li>Quality trimming</li>
<li>Applying digital normalization</li>
<li>Running the actual assembly</li>
<li>BLASTing your assembly</li>
</ol>
<h4 id="&#36136;&#37327;&#20462;&#21098;&#21644;&#24207;&#21015;&#36807;&#28388;">1. &#36136;&#37327;&#20462;&#21098;&#21644;&#24207;&#21015;&#36807;&#28388;</h4>
<ul>
<li>&#23433;&#35013;kmer</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">cd</span> <span class="ot">${HOME}</span>
<span class="kw">mkdir</span> -p projects/eelpond
<span class="kw">python2.7</span> -m virtualenv projects/eelpond/env
<span class="kw">source</span> <span class="ot">${HOME}</span>/projects/eelpond/env/bin/activate
<span class="kw">mkdir</span> -p src
<span class="kw">cd</span> src
<span class="kw">git</span> clone --branch v1.3 https://github.com/ged-lab/khmer.git
<span class="kw">cd</span> khmer
<span class="kw">make</span> install</code></pre></div>
<p>&#20854;&#20013;&#65292;&#36873;&#39033;<code>virtualenv</code>&#20351;&#25105;&#20204;&#21487;&#20197;&#22312;&#27809;&#26377;root&#26435;&#38480;&#26465;&#20214;&#19979;&#23433;&#35013;python&#36719;&#20214;&#12290;</p>
<ul>
<li>&#19979;&#36733;&#25968;&#25454;</li>
</ul>
<p><a href="https://darchive.mblwhoilibrary.org/bitstream/handle/1912/5613/0Hour_ATCACG_L002_R1_001.fastq.gz?sequence=2&amp;isAllowed=y">0Hour_ATCACG_L002_R1_001.fastq.gz</a></p>
<p>&#20026;&#20102;&#33410;&#30465;&#26102;&#38388;&#65292;&#20808;&#19981;&#35201;&#20351;&#29992;&#25972;&#20010;&#25968;&#25454;&#36827;&#34892;&#36816;&#34892;&#65292;&#21487;&#20197;&#20808;&#25552;&#21462;&#37096;&#20998;&#25968;&#25454;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">cd</span> raw
<span class="kw">ln</span> -fs <span class="ot">${HOME}</span>/data/nemo/*.fastq.gz .
<span class="kw">mkdir</span> -p extract
<span class="kw">for</span> <span class="kw">file</span> in raw/*.fastq.gz
<span class="kw">do</span>
    <span class="kw">gunzip</span> -c <span class="ot">${file}</span> <span class="kw">|</span> <span class="kw">head</span> -400000 <span class="kw">|</span> <span class="kw">gzip</span> \
        <span class="kw">&gt;</span> extract/<span class="ot">${file%%</span>.fastq.gz<span class="ot">}</span>.extract.fastq.gz
<span class="kw">done</span></code></pre></div>
<ul>
<li>&#20351;&#29992;FastQC&#35780;&#20272;&#36136;&#37327;</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">fastqc</span> --threads 1 *.fastq.gz --outdir=<span class="ot">${HOME}</span>/fastqc</code></pre></div>
<ul>
<li>&#25214;&#21040;&#27491;&#30830;&#30340;Illumina adapters</li>
</ul>
<p>&#38656;&#35201;&#30830;&#35748;&#26159;&#21542;&#37319;&#29992;&#20102;&#27491;&#30830;&#30340;adaptor&#65292;&#22914;&#26524;&#19981;&#26159;&#27491;&#30830;&#30340;adaptor&#65292;&#37027;&#20040;read&#23558;&#19981;&#20250;&#34987;&#20462;&#21098;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">wget</span> https://sources.debian.net/data/main/t/trimmomatic/0.32+dfsg-2/adapters/TruSeq3-PE.fa</code></pre></div>
<ul>
<li>&#23545;&#27599;&#23545;&#25991;&#20214;&#36827;&#34892;adapter&#20462;&#21098;&#65306;</li>
</ul>
<p>&#23545;&#20197;&#19979;&#20004;&#20010;paired-end&#25991;&#20214;&#36827;&#34892;&#22788;&#29702;&#65306;</p>
<pre><code>24HourB_GCCAAT_L002_R1_001.fastq.gz
24HourB_GCCAAT_L002_R2_001.fastq.gz</code></pre>
<p>&#23545;&#20108;&#32773;&#20998;&#21035;&#31616;&#20889;&#20026;<R1 file>&#21644;<R2 file>, &#19988;&#38656;&#35201;&#25351;&#23450;&#19968;&#20010;&#29305;&#23450;&#30340;<sample name>&#12290;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># run trimmomatic</span>
<span class="kw">trimmomatic</span> PE <span class="kw">&lt;</span>R1 FILE<span class="kw">&gt;</span> <span class="kw">&lt;</span>R2 FILE<span class="kw">&gt;</span> s1_pe s1_se s2_pe s2_se \
    ILLUMINACLIP:<span class="ot">${HOME}</span>/projects/eelpond/TruSeq3-PE.fa:2:30:10

<span class="co"># interleave the remaining paired-end files</span>
<span class="kw">interleave-reads.py</span> s1_pe s2_pe <span class="kw">|</span> <span class="kw">gzip</span> -9c \
    <span class="kw">&gt;</span> ../trimmed/<span class="kw">&lt;</span>SAMPLE NAME<span class="kw">&gt;</span>.pe.fq.gz

<span class="co"># combine the single-ended files</span>
<span class="kw">cat</span> s1_se s2_se <span class="kw">|</span> <span class="kw">gzip</span> -9c <span class="kw">&gt;</span> ../trimmed/<span class="kw">&lt;</span>SAMPLE NAME<span class="kw">&gt;</span>.se.fq.gz

<span class="co"># clear the temporary files</span>
<span class="kw">rm</span> *

<span class="co"># make it hard to delete the files you just created</span>
<span class="kw">cd</span> ../trimmed
<span class="kw">chmod</span> u-w *</code></pre></div>
<ul>
<li>&#36890;&#36807;&#33050;&#26412;&#23454;&#29616;&#33258;&#21160;&#21270;</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">python</span> <span class="ot">${HOME}</span>/src/khmer/sandbox/write-trimmomatic.py <span class="kw">&gt;</span> trim.sh
<span class="kw">more</span> trim.sh
<span class="kw">bash</span> trim.sh</code></pre></div>
<ul>
<li>&#36136;&#37327;&#20462;&#21098;</li>
</ul>
<p>&#23545;&#20110;&#25152;&#26377;&#30340;<code>.pe.fq.gz</code>&#21644;<code>.se.fq.gz</code>&#25991;&#20214;&#65292;&#36816;&#34892;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">gunzip</span> -c <span class="kw">&lt;</span>filename<span class="kw">&gt;</span> <span class="kw">|</span> <span class="kw">fastq_quality_filter</span> -Q33 -q 30 -p 50 <span class="kw">|</span> <span class="kw">gzip</span> -9c \
<span class="kw">&gt;</span> <span class="kw">&lt;</span>filename<span class="kw">&gt;</span>.qc.fq.gz</code></pre></div>
<p>&#27492;&#22806;&#65292;&#21487;&#20197;&#23558;&#21629;&#20196;<code>fastx_trimmer -Q33 -l 70 |</code>&#25918;&#20837;&#20854;&#20013;&#23454;&#29616;&#22266;&#23450;&#38271;&#24230;read&#30340;&#36755;&#20986;&#12290;&#21478;&#22806;&#65292;&#22914;&#26524;&#31227;&#38500;<code>-Q33</code>&#36873;&#39033;&#65292;&#37027;&#20040;&#21487;&#33021;&#23548;&#33268;&#36136;&#37327;&#20540;&#38382;&#39064;&#12290;</p>
<ul>
<li>&#33258;&#21160;&#21270;</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">for</span> <span class="kw">file</span> in *
<span class="kw">do</span>
     <span class="kw">echo</span> working with <span class="ot">${file}</span>
     <span class="ot">newfile=${file%%</span>.fq.gz<span class="ot">}</span>.qc.fq.gz
     <span class="kw">gunzip</span> -c <span class="ot">${file}</span> <span class="kw">|</span> <span class="kw">fastq_quality_filter</span> -Q33 -q 30 -p 50 <span class="kw">|</span> <span class="kw">gzip</span> -9c \
         <span class="kw">&gt;</span> ../filtered/<span class="ot">${newfile}</span>
<span class="kw">done</span></code></pre></div>
<ul>
<li>&#20174;&#38388;&#38548;&#25991;&#20214;&#20013;&#25552;&#21462;paired-ends</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> filtered-pairs
<span class="kw">cd</span> filtered-pairs
<span class="kw">for</span> <span class="kw">file</span> in ../filtered/*.pe.qc.fq.gz
<span class="kw">do</span>
   <span class="kw">extract-paired-reads.py</span> <span class="ot">${file}</span>
<span class="kw">done</span></code></pre></div>
<ul>
<li>&#32467;&#26463;</li>
</ul>
<p>&#26368;&#21518;&#24471;&#21040;&#30340;&#25991;&#20214;&#21015;&#34920;&#65306;</p>
<pre><code>raw/24HourB_GCCAAT_L002_R1_001.fastq.gz                   - the original data
raw/24HourB_GCCAAT_L002_R2_001.fastq.gz
trimmed/24HourB_GCCAAT_L002_R1_001.pe.fq.gz               - adapter trimmed pe
trimmed/24HourB_GCCAAT_L002_R1_001.se.fq.gz               - adapter trimmed orphans
filtered/24HourB_GCCAAT_L002_R1_001.pe.qc.fq.gz           - FASTX filtered
filtered/24HourB_GCCAAT_L002_R1_001.se.qc.fq.gz           - FASTX filtered orphans
filtered-pairs/24HourB_GCCAAT_L002_R1_001.pe.qc.fq.gz.pe  - FASTX filtered PE
filtered-pairs/24HourB_GCCAAT_L002_R1_001.pe.qc.fq.gz.se  - FASTX filtered SE</code></pre>
<p>&#21024;&#38500;&#26080;&#29992;&#25991;&#20214;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">#Well, first, you can get rid of the original data. You already have it on a disk somewhere, right?</span>

<span class="kw">rm</span> raw/*
<span class="kw">rmdir</span> raw

<span class="co">#Next, you can get rid of the trimmed files, since you only want the QC files. So</span>

<span class="kw">rm</span> -f trimmed/*
<span class="kw">rmdir</span> trimmed

<span class="co">#And, finally, you can toss the filtered files, because you&#8217;ve turned those into *.pe and *.se files:</span>

<span class="kw">rm</span> filtered/*
<span class="kw">rmdir</span> filtered</code></pre></div>
<p>&#26368;&#21518;&#21097;&#20313;&#25991;&#20214;&#65306;</p>
<pre><code>filtered-pairs/24HourB_GCCAAT_L002_R1_001.pe.qc.fq.gz.pe   - FASTX filtered PE
filtered-pairs/24HourB_GCCAAT_L002_R1_001.pe.qc.fq.gz.se   - FASTX filtered SE
filtered-pairs/24HourB_GCCAAT_L002_R1_001.se.qc.fq.gz      - FASTX filtered orphans</code></pre>
<p>&#27880;&#24847;&#65306;</p>
<ol style="list-style-type: decimal">
<li>&#25991;&#20214;&#21517;&#35201;&#26041;&#20415;&#26174;&#31034;&#20854;&#20449;&#24687;&#19982;&#21382;&#21490;&#65292;&#21363;&#20351;&#19985;&#38475;&#20063;&#27809;&#20851;&#31995;</li>
<li>&#27599;&#19968;&#27493;&#37117;&#21487;&#20197;&#24314;&#31435;&#19968;&#20010;&#26032;&#30340;&#30446;&#24405;&#65292;&#26041;&#20415;&#22312;&#21518;&#32493;&#27493;&#39588;&#20013;&#21024;&#38500;&#26080;&#29992;&#25991;&#20214;</li>
</ol>
<p>&#37325;&#21629;&#21517;&#25991;&#20214;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">## PE</span>
<span class="kw">for</span> <span class="kw">file</span> in filtered-pairs/*.pe
<span class="kw">do</span>
   <span class="ot">newfile=${file%%</span>.pe.qc.fq.gz.pe<span class="ot">}</span>.pe.qc.fq
   <span class="kw">mv</span> <span class="ot">$file</span> <span class="ot">$newfile</span>
   <span class="kw">gzip</span> <span class="ot">$newfile</span>
<span class="kw">done</span>

<span class="co">##SE</span>
<span class="kw">for</span> <span class="kw">file</span> in filtered-pairs/*.se
<span class="kw">do</span>
  <span class="ot">otherfile=${file%%</span>.pe.qc.fq.gz.se<span class="ot">}</span>.se.qc.fq.gz <span class="co"># the orphans</span>
  <span class="kw">gunzip</span> -c <span class="ot">${otherfile}</span> <span class="kw">&gt;</span> combine
  <span class="kw">cat</span> <span class="ot">${file}</span> <span class="kw">&gt;&gt;</span> combine
  <span class="kw">gzip</span> -c combine <span class="kw">&gt;</span> <span class="ot">${otherfile}</span> <span class="co"># now all the single reads together</span>
  <span class="kw">rm</span> <span class="ot">${file}</span> combine
<span class="kw">done</span>

<span class="co">##make the end product files read-only</span>
<span class="kw">chmod</span> u-w filtered-pairs/*</code></pre></div>
<h4 id="&#25968;&#23383;&#26631;&#20934;&#21270;&#24212;&#29992;">&#25968;&#23383;&#26631;&#20934;&#21270;&#24212;&#29992;</h4>
<ul>
<li>&#36830;&#25509;&#25968;&#25454;</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> -p <span class="ot">${HOME}</span>/filtered-pairs
<span class="kw">ln</span> -fs <span class="ot">${HOME}</span>/data/*.qc.fq.gz <span class="ot">${HOME}</span>/filtered-pairs/</code></pre></div>
<ul>
<li>&#36816;&#34892;&#25968;&#23383;&#26631;&#20934;&#21270;</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
<span class="co">##paired-end reads</span>
<span class="kw">cd</span> <span class="ot">${HOME}</span>
<span class="kw">mkdir</span> diginorm
<span class="kw">cd</span> diginorm
<span class="kw">normalize-by-median.py</span> --paired -ksize 20 --cutoff 20 -n_tables 4 \
  --min-tablesize 3e8 --savetable normC20k20.ct \
  ../filtered-pairs/*.pe.qc.fq.gz

<span class="co">##single-end reads</span>
<span class="kw">normalize-by-median.py</span> --cutoff 20 --loadtable normC20k20.ct \
  --savetable normC20k20.ct ../filtered-pairs/*.se.qc.fq.gz</code></pre></div>
<p>&#20854;&#20013;&#65292;<code>--paired</code>&#34920;&#31034;&#22788;&#29702;PE&#25968;&#25454;&#65292;&#20445;&#35777;&#25152;&#26377;reads&#19981;&#20250;&#33853;&#21333;&#65307;&#32780;<code>--n_tables</code>&#21644;<code>--min_tablesize</code>&#21442;&#25968;&#21017;&#25351;&#23450;&#22810;&#23569;&#20869;&#23384;&#21487;&#20351;&#29992;&#65292;&#24212;&#27604;&#35745;&#31639;&#26426;&#20869;&#23384;&#23567;&#12290;&#20363;&#22914;&#65292;&#23545;&#20110;&#36716;&#24405;&#32452;&#26469;&#35828;&#65292;&#26368;&#22909;&#20026;~60GB&#24038;&#21491;&#65292;&#25152;&#20197;&#21487;&#20197;&#37319;&#29992;<code>--n_tables 4 --min_tablesize 15e9</code>; &#32780;&#23545;&#20110;&#25968;&#30334;&#30334;&#19975;&#30340;reads&#65292;16GB&#26159;&#36275;&#22815;&#30340;&#12290;&#65288;&#35814;&#35265;http://khmer.readthedocs.org/en/latest/choosing-hash-sizes.html&#65289;</p>
<ul>
<li>&#21098;&#25481;&#21487;&#33021;&#38169;&#35823;&#30340;k-mers</li>
</ul>
<p>&#36941;&#21382;&#25152;&#26377;reads&#24182;&#21098;&#25481;&#39640;&#35206;&#30422;reads&#20013;&#30340;&#20302;&#20016;&#24230;&#37096;&#20998;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> abundfilt
<span class="kw">cd</span> abundfilt
<span class="kw">filter-abund.py</span> --variable-coverage ../diginorm/normC20k20.ct \
  --threads <span class="ot">${THREADS:-</span>1<span class="ot">}</span> ../diginorm/*.keep</code></pre></div>
<p>&#36825;&#19968;&#27493;&#20250;&#20351;&#19968;&#37096;&#20998;reads&#33853;&#21333;&#65292;&#20294;&#26159;&#27809;&#20851;&#31995;&#65292;&#22240;&#20026;&#23427;&#20204;&#30340;partner read&#19981;&#22909;&#12290;</p>
<ul>
<li>&#37325;&#21629;&#21517;&#25991;&#20214;</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co"># break out the orphaned and still-paired reads:</span>
<span class="kw">mkdir</span> digiresult
<span class="kw">cd</span> digiresult
<span class="kw">for</span> <span class="kw">file</span> in ../abundfilt/*.pe.*.abundfilt
<span class="kw">do</span>
   <span class="kw">extract-paired-reads.py</span> <span class="ot">${file}</span>
<span class="kw">done</span>

<span class="co"># combine the orphaned reads into a single file:</span>
<span class="kw">for</span> <span class="kw">file</span> in *.se.qc.fq.gz.keep.abundfilt
<span class="kw">do</span>
   <span class="ot">pe_orphans=${file%%</span>.se.qc.fq.gz.keep.abundfilt<span class="ot">}</span>.pe.qc.fq.gz.keep.abundfilt.se
   <span class="ot">newfile=${file%%</span>.se.qc.fq.gz.keep.abundfilt<span class="ot">}</span>.se.qc.keep.abundfilt.fq.gz
   <span class="kw">cat</span> <span class="ot">${file}</span> ../digiresult/<span class="ot">${pe_orphans}</span> <span class="kw">|</span> <span class="kw">gzip</span> -c <span class="kw">&gt;</span> ../digiresult/<span class="ot">${newfile}</span>
   <span class="kw">rm</span> <span class="ot">${pe_orphans}</span>
<span class="kw">done</span>

<span class="co"># rename the remaining PE reads &amp; compress those files:</span>
<span class="kw">for</span> <span class="kw">file</span> in *.abundfilt.pe
<span class="kw">do</span>
   <span class="ot">newfile=${file%%</span>.fq.gz.keep.abundfilt.pe<span class="ot">}</span>.keep.abundfilt.fq
   <span class="kw">mv</span> <span class="ot">${file}</span> <span class="ot">${newfile}</span>
   <span class="kw">gzip</span> <span class="ot">${newfile}</span>
<span class="kw">done</span></code></pre></div>
<p>&#21097;&#20313;&#25991;&#20214;&#65306;</p>
<pre><code>filtered-reads/6Hour_CGATGT_L002_R1_005.pe.qc.fq.gz
filtered-reads/6Hour_CGATGT_L002_R1_005.se.qc.fq.gz
diginorm/6Hour_CGATGT_L002_R1_005.pe.qc.fq.gz.keep
diginorm/6Hour_CGATGT_L002_R1_005.se.qc.fq.gz.keep
diginorm/normC20k20.ct
abundfilt/6Hour_CGATGT_L002_R1_005.pe.qc.fq.gz.keep.abundfilt
abundfilt/6Hour_CGATGT_L002_R1_005.se.qc.fq.gz.keep.abundfilt
digiresult/6Hour_CGATGT_L002_R1_005.pe.qc.keep.abundfilt.fq.gz
digiresult/6Hour_CGATGT_L002_R1_005.se.qc.keep.abundfilt.fq.gz</code></pre>
<h4 id="&#36827;&#34892;&#27491;&#24335;&#30340;assembly">3. &#36827;&#34892;&#27491;&#24335;&#30340;assembly</h4>
<p>&#19979;&#38754;&#36825;&#20123;&#37117;&#20250;&#22312;screen&#20013;&#36816;&#34892;&#65292;&#33267;&#23569;&#26377;15GB&#20869;&#23384;</p>
<ul>
<li>&#23433;&#35013;Trinity</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">wget</span> https://github.com/trinityrnaseq/trinityrnaseq/archive/v2.0.4.tar.gz \
  -O trinity.tar.gz
<span class="kw">tar</span> xzf trinity.tar.gz
<span class="kw">cd</span> trinityrnaseq*/
<span class="kw">make</span> <span class="kw">|&amp;</span> <span class="kw">tee</span> trinity-build.log</code></pre></div>
<ul>
<li>&#21019;&#24314;&#36827;&#34892;assemble&#30340;&#25991;&#20214;</li>
</ul>
<p>&#23545;&#20110;PE&#25968;&#25454;&#65292;Trinity&#38656;&#35201;&#20004;&#20010;&#25991;&#20214;: left &amp; right, &#23427;&#20204;&#21487;&#20197;&#21547;&#26377;&#33853;&#21333;reads&#12290;&#25152;&#20197;&#21487;&#20197;&#23558;&#38388;&#38548;&#30340;pe&#25991;&#20214;&#25286;&#20998;&#20026;&#20004;&#20010;&#65292;&#24182;&#23558;se&#24207;&#21015;&#28155;&#21152;&#36827;&#21435;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">for</span> <span class="kw">file</span> in *.pe.qc.keep.abundfilt.fq.gz
<span class="kw">do</span>
   <span class="kw">split-paired-reads.py</span> <span class="ot">${file}</span>
<span class="kw">done</span>

<span class="kw">cat</span> *.1 <span class="kw">&gt;</span> left.fq
<span class="kw">cat</span> *.2 <span class="kw">&gt;</span> right.fq

<span class="kw">gunzip</span> -c *.se.qc.keep.abundfilt.fq.gz <span class="kw">&gt;&gt;</span> left.fq</code></pre></div>
<ul>
<li>&#20351;&#29992;Trinity&#36827;&#34892;assembly&#65306;</li>
</ul>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">Trinity</span> --left left.fq --right right.fq --seqTypr fq --max_memory 14G --CPU 1</code></pre></div>
<p>&#26368;&#32456;&#32463;&#36807;&#38271;&#26102;&#38388;&#30340;&#36816;&#34892;&#32467;&#26463;&#21518;&#65292;&#19968;&#20010;&#32452;&#35013;&#22909;&#30340;&#36716;&#24405;&#32452;&#23601;&#20250;&#20135;&#29983;<code>Trinity.fasta</code>&#12290;</p>
<h4 id="&#32452;&#35013;&#25968;&#25454;&#30340;blast">4. &#32452;&#35013;&#25968;&#25454;&#30340;blast</h4>
<p>&#23433;&#35013;blastkit&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">git</span> clone https://github.com/ctb/blastkit.git -b ec2
<span class="kw">cd</span> blastkit/www
<span class="kw">ln</span> -fs <span class="ot">$PWD</span> /var/www/blastkit

<span class="kw">mkdir</span> files
<span class="kw">chmod</span> a+rxwt files

<span class="kw">python</span> check.py</code></pre></div>
<p>&#28155;&#21152;&#25968;&#25454;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">cp</span> Trinity.fasta /blastkie/db/db.fa

<span class="co">#&#25110;&#32773;&#20351;&#29992;&#24050;&#26377;&#25968;&#25454;</span>
<span class="kw">curl</span> -O https://s3.amazonaws.com/public.ged.msu.edu/trinity-nematostella-raw.fa.gz
<span class="kw">gunzip</span> trinity-nematostella-raw.fa.gz
<span class="kw">mv</span> trinity-nematostella-raw.fa db/db.fa</code></pre></div>
<p>&#26684;&#24335;&#21270;&#25968;&#25454;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">formatdb</span> -i db/db.fa -o T -p F
<span class="kw">python</span> index-db.py db/db.fa</code></pre></div>
<p>&#36816;&#34892;blastkit&#65306; &#20351;&#29992;&#20197;&#19979;&#24207;&#21015;</p>
<pre><code>CAGCCTTTAGAAGGAAACAGTGGCAATATATAATTCTAGATGAAGCTCAGAATATCAAAA
ATTTTAAAAGTCAAAGGTGGCAGTTGCTGTTGAATTTTTCAAGTCAGAGGAGACTTTTGT
TGACTGGAACACCTTTGCAGAACAATTTGATGGAGCTGTGGTCGCTTATGCATTTCCTCA
TGCCATCAATGTTTGCTTCTCATAAAGATTTTAGGGAGTGGTTTTCTAACCCTGTTACAG
GGATGATTGAAGGGAATTCAG</code></pre>
<p>&#23427;&#24212;&#35813;&#20250;&#22312;&#20320;&#30340;&#32452;&#35013;&#20013;&#21305;&#37197;&#21040;&#19968;&#20123;&#25968;&#25454;&#12290;</p>
<h4 id="&#26500;&#24314;&#36716;&#24405;&#23478;&#26063;">5. &#26500;&#24314;&#36716;&#24405;&#23478;&#26063;</h4>
<p>&#24314;&#31435;&#20010;&#20154;&#30340;&#20108;&#36827;&#21046;&#31243;&#24207;&#25991;&#20214;&#22841;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> -p <span class="ot">${HOME}</span>/bin
<span class="kw">export</span> <span class="ot">PATH=${PATH}</span>:<span class="ot">${HOME}</span>/bin
<span class="kw">echo</span> <span class="st">&#39;export PATH=${PATH}:${HOME}/bin&#39;</span> <span class="kw">&gt;&gt;</span> <span class="ot">${HOME}</span>/.bashrc</code></pre></div>
<p>&#25335;&#36125;&#25968;&#25454;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">gzip</span> -c trinity_out_dir/Trinity.fasta <span class="kw">&gt;</span> trinity-nematostella-raw.fa.gz</code></pre></div>
<p>&#36816;&#34892;khmer&#20998;&#38548;&#65306;</p>
<p>Partioning&#36816;&#34892;&#22522;&#20110;de Bruijin graph&#30340;&#25104;&#31751;&#31639;&#27861;&#21487;&#20197;&#36890;&#36807;&#36807;&#28193;&#24207;&#21015;&#37325;&#21472;&#23558;&#20320;&#30340;&#36716;&#24405;&#26412;&#31751;&#21270;&#12290;&#20063;&#23601;&#26159;&#65292;&#23558;&#36716;&#24405;&#26412;&#20998;&#32452;&#20026;&#36716;&#24405;&#26412;&#23478;&#26063;&#65288;&#26681;&#25454;shared&#24207;&#21015;&#65289;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> partitions
<span class="kw">cd</span> partitions
<span class="kw">do-partition.py</span> -x 1e9 -N 4 --threads 1 nema \
  ../trinity-Nematostella-raw.fa.gz</code></pre></div>
<p>&#36807;10&#22810;&#20998;&#38047;&#21518;&#65292;&#36755;&#20986;&#30340;&#25991;&#20214;&#65288;&#20197;.part&#32467;&#23614;&#65289;&#21253;&#21547;&#26377;&#20998;&#21306;&#20998;&#37197;&#20449;&#24687;&#12290;&#21487;&#20197;&#20998;&#32452;&#21644;&#37325;&#21629;&#21517;&#36825;&#20123;&#24207;&#21015;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">rename-with-partitions.py</span> nema trinity-nematostella-raw.fa.gz.part
<span class="kw">mv</span> trinit-nematostella-raw.fa.gz.part.renamed.fasta.gz \
  trinity-nematostella.renamed.fa.gz</code></pre></div>
<p>&#26597;&#30475;&#37325;&#21629;&#21517;&#24207;&#21015;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">gunzip</span> -c trinity-nematostella.renamed.fa.gz <span class="kw">|</span> <span class="kw">head</span></code></pre></div>
<p>&#27599;&#26465;&#24207;&#21015;&#22823;&#27010;&#20855;&#26377;&#36825;&#26679;&#30340;&#21517;&#23383;&#65306;</p>
<pre><code>&gt;nema.id1.tr16001 1_of_1_in_tr16001 len=261 id=1 tr=16001</code></pre>
<p>&#35299;&#37322;&#65306;</p>
<ul>
<li>nema - &#37325;&#21629;&#21517;&#26102;&#35774;&#23450;&#30340;&#21069;&#32512;</li>
<li>idN - &#24207;&#21015;&#30340;&#29420;&#26377;id</li>
<li>trN - &#26159;&#36716;&#24405;&#26412;&#23478;&#26063;</li>
<li>1_0f_1_in_tr16001 - &#34920;&#31034;&#35813;&#36716;&#24405;&#26412;&#23478;&#26063;&#20165;&#26377;&#19968;&#20010;&#25104;&#21592;</li>
<li>len - &#24207;&#21015;&#38271;&#24230;</li>
</ul>
<h4 id="&#27880;&#37322;&#36716;&#24405;&#26412;&#23478;&#26063;">6. &#27880;&#37322;&#36716;&#24405;&#26412;&#23478;&#26063;</h4>
<p>&#29983;&#25104;&#25152;&#38656;&#25991;&#20214;&#65288;&#19978;&#33410;&#65289;&#25110;&#19979;&#36733;&#20043;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">curl</span> -O http://public.ged.msu.edu.s3.amazonaws.com/trinity-nematostella.renamed.fa.gz</code></pre></div>
<p>&#30001;&#20110;BLASTs&#23558;&#33457;&#36153;&#36739;&#38271;&#26102;&#38388;&#65292;&#25152;&#20197;&#21487;&#20197;&#20351;&#29992;canned BLASTs&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">cd</span> <span class="ot">${HOME}</span>/annotation
<span class="kw">curl</span> -O http://public.ged.msu.edu.s3.amazonaws.com/nema.x.mouse.gz
<span class="kw">curl</span> -O http://public.ged.msu.edu.s3.amazonaws.com/mouse.x.nema.gz
<span class="kw">gunzip</span> nema.x.mouse.gz
<span class="kw">gunzip</span> mouse.x.nema.gz</code></pre></div>
<p>&#38024;&#23545;&#23567;&#40736;&#30340;&#21021;&#27493;&#27880;&#37322;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">#&#20551;&#23450;&#21516;&#28304;&#24615;&#65292;&#20570;BLASTs&#21644;&#30456;&#20114;&#26368;&#20339;&#21305;&#37197;&#20998;&#26512;&#65292;&#39318;&#20808;&#35299;&#21387;&#25991;&#20214;</span>
<span class="kw">gunzip</span> -c ../partitions/trinity-nematostella.renamed.fa.gz \
  trinity-nematostella.renamed.fa

<span class="co">#&#33719;&#21462;&#26368;&#26032;mouse RefSeq</span>
<span class="kw">for</span> <span class="kw">file</span> in mouse.1.protein.faa.gz mouse.2.protein.faa.gz
<span class="kw">do</span>
     <span class="kw">curl</span> -O ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/<span class="ot">${file}</span>
<span class="kw">done</span>
<span class="kw">gunzip</span> mouse.[123].protein.faa.gz
<span class="kw">cat</span> mouse.[123].protein.faa <span class="kw">&gt;</span> mouse.protein.faa

<span class="co">#&#26684;&#24335;&#21270;blast&#24211;</span>
<span class="kw">formatdb</span> -i mouse.protein.faa -o T -p T
<span class="kw">formatdb</span> -i trinity-nematostella.renamed.fa -o T -p F

<span class="co">#&#22312;screen&#20013;&#36827;&#34892;&#21452;&#21521;blast</span>
<span class="kw">blastx</span> -db mouse.protein.faa -query trinity-nematostella.renamed.fa \
   -evalue 1e-3 -num_threads 8 -num_descriptions 4 -num_alignments 4 \
   -out nema.x.mouse
   <span class="kw">tblastn</span> -db trinity-nematostella.renamed.fa -query mouse.protein.faa \
   -evalue 1e-3 -num_threads 8 -num_descriptions 4 -num_alignments 4 \
   -out mouse.x.nema</code></pre></div>
<p>&#25351;&#23450;&#24207;&#21015;&#21517;&#31216;&#65306;&#26681;&#25454;&#30456;&#20114;&#26368;&#20339;&#21305;&#37197;&#35745;&#31639;&#20551;&#23450;&#30340;homology (best BLAST hit) and orthology (reciprocal best hits):</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">make-uni-best-hits.py</span> nema.x.moutse nema.x.mouse.homol
<span class="kw">make-reciprocal-best-hits.py</span> nema.x.mouse mouse.x.nema.mouse.ortho

<span class="co">#&#20934;&#22791;mouse&#20449;&#24687;&#20013;&#30340;&#19968;&#20123;</span>
<span class="kw">make-namedb.py</span> mouse.protein.faa mouse.namedb
<span class="kw">python</span> -m screed.fadbm mouse.protein.faa

<span class="co">#&#26368;&#21518;&#27880;&#37322;&#24207;&#21015;</span>
<span class="kw">annotate-seqs.py</span> trinity-nematostella.renamed.fa nema.x.mouse.ortho \
  nema.x.mouse.homol</code></pre></div>
<p>&#26368;&#21518;&#22238;&#30475;&#21040;&#65306;</p>
<pre><code>207533 sequences total
10471 annotated / ortho
95726 annotated / homol
17215 annotated / tr
123412 total annotated</code></pre>
<blockquote>
<p>&#22914;&#26524;&#20854;&#20013;&#20219;&#20309;&#19968;&#20010;&#25968;&#20026;0&#65292;&#37027;&#20040;&#38656;&#35201;&#37325;&#26032;&#20570;BLAST&#12290;</p>
</blockquote>
<p>&#26368;&#21518;&#29983;&#25104;&#25991;&#20214;<code>trinity-nematostella.renamed.fa.annot</code>&#65292;&#20854;&#20013;&#30340;&#24207;&#21015;&#20284;&#19979;&#65306;</p>
<pre><code>&gt;nematostella.id1.tr115222 h=43% =&gt; suppressor of tumorigenicity 7 protein isoform 2 [Mus musculus] 1_of_7_in_tr115222 len=1635 id=1 tr=115222 1_of_7_in_tr115222 len=1635 id=1 tr=115222</code></pre>
<p>&#24314;&#35758;&#37325;&#21629;&#21517;&#25913;&#20026;&#35813;&#25991;&#20214;&#20026;<code>nematostella.fa</code>,&#29992;&#20854;&#36827;&#34892;4&#20013;&#30340;blast&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">cp</span> trinity-nematostella.renamed.fa.anoot Nematostella.fa</code></pre></div>
<p>&#27880;&#37322;&#36807;&#31243;&#36824;&#20250;&#20135;&#29983;2&#20010;CSV&#25991;&#20214;&#65292;&#23567;&#30340;&#20026;<code>trinity-nematostella.renamed.fa.annot.csv</code>&#65292;&#21253;&#21547;&#36830;&#25509;&#21040;&#21516;&#28304;&#20449;&#24687;&#30340;&#24207;&#21015;&#21517;&#23383;&#65292;&#32780;&#22823;&#30340;&#37027;&#20010;<code>trinity-nematostella.renamed.fa.annot.large.csv</code>&#21017;&#21253;&#21547;&#25152;&#26377;&#20449;&#24687;&#12290;</p>
<h4 id="&#34920;&#36798;&#20998;&#26512;">7. &#34920;&#36798;&#20998;&#26512;</h4>
<p>&#39318;&#20808;&#65292;&#33719;&#21462;&#20998;&#21106;&#21644;&#37325;&#21629;&#21517;&#25968;&#25454;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">curl</span> -O http://athyra.idyll.org/~t/trinity-nematostella.renamed.fa.gz
<span class="kw">gunzip</span> -c trinity-nematostella.renamed.fa.gz <span class="kw">&gt;</span> nematostella.fa</code></pre></div>
<ul>
<li>&#23433;&#35013;rsem</li>
</ul>
<p>&#20351;&#29992;<a href="http://deweylab.biostat.wisc.edu/rsem/">RSEM</a>&#36827;&#34892;&#34920;&#36798;&#20998;&#26512;&#65292;&#20351;&#29992;<a href="http://www.biostat.wisc.edu/~kendzior/EBSEQ/">EBSeq</a>&#36827;&#34892;&#24046;&#24322;&#34920;&#36798;&#20998;&#26512;&#65292;&#23433;&#35013;&#26041;&#27861;&#22914;&#19979;&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">curl</span> -O http://deweylab.biostat.wisc.edu/rsem/src/rsem-1.2.8.tar.gz
<span class="kw">tar</span> xzf rsem-1.2.8.tar.gz
<span class="kw">cd</span> rsem-1.2.8
<span class="kw">make</span>
<span class="kw">cd</span> EBSeq
<span class="kw">make</span>

<span class="co">#RSEM&#38656;&#35201;bowtie</span>
<span class="kw">curl</span> -O -L http://sourceforge.net/projects/bowtie-bio/files/bowtie/0.12.7/bowtie-0.12.7-linux-x86_64.zip
<span class="kw">unzip</span> bowtie-0.12.7-linux-x86_64.zip
<span class="kw">cd</span> bowtie-0.12.7
<span class="kw">cp</span> bowtie bowtie-build bowtie-inspect /usr/local/bin</code></pre></div>
<p>&#20934;&#22791;reference&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">mkdir</span> rsem
<span class="kw">cd</span> rsem

<span class="kw">ln</span> -fs ../nematostella.fa .

<span class="co">#make a transcript-to-gene-map file</span>
<span class="kw">python</span> make-transcript-to-gene-map-file.py nematostella.fa nematostella.fa.tr_to_genes

<span class="co">#ask rsen&#20934;&#22791;reference</span>
<span class="kw">rsem-prepare-reference</span> --transcript-to-gene-map nematostella.fa.tr_to_genes nematostella.fa nema</code></pre></div>
<p>&#25214;&#21040;&#24182;&#21015;&#20986;reads</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="co">#&#25214;&#21040;QC reads&#65292;&#38142;&#25509;&#36807;&#26469;</span>
<span class="kw">ln</span> -fs /data/*.pe.qc.fq.gz
<span class="kw">ls</span> -l *.pe.qc.fq.gz <span class="kw">&gt;</span> list.txt</code></pre></div>
<p>&#36816;&#34892;RSEM</p>
<p>&#23545;&#20110;<code>list.txt</code>&#20013;&#30340;&#27599;&#20010;&#25991;&#20214;&#36816;&#34892;RSEM(&#27604;&#36739;&#32791;&#26102;&#38388;&#65292;&#22312;screen&#20013;&#36827;&#34892;)&#65306;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="ot">n=</span>1
<span class="kw">for</span> <span class="kw">filename</span> in <span class="ot">$(</span><span class="kw">cat</span> list.txt<span class="ot">)</span>
<span class="kw">do</span>
    <span class="kw">echo</span> mapping <span class="ot">$filename</span>
    <span class="kw">gunzip</span> -c <span class="ot">$filename</span> <span class="kw">&gt;</span> <span class="ot">${n}</span>.fq
    <span class="kw">/usr/local/share/khmer/scripts/split-paired-reads.py</span> <span class="ot">${n}</span>.fq
    <span class="kw">rsem-calculate-expression</span> --paired-end <span class="ot">${n}</span>.fq.1 <span class="ot">${n}</span>.fq.2 nema -p 4 <span class="ot">${n}</span>.fq
    <span class="kw">rm</span> <span class="ot">${n}</span>.fq <span class="ot">${n}</span>.fq.[12] <span class="ot">${n}</span>.fq.transcript.bam <span class="ot">${n}</span>.fq.transcript.sorted.bam
    <span class="ot">n=$(($n</span> + 1<span class="ot">))</span>
<span class="kw">done</span></code></pre></div>
<p>&#32858;&#38598;&#32467;&#26524;</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">rsem-generate-data-matrix</span> [0-9].fq.genes.results 10.fq.genes.results <span class="kw">&gt;</span> 0-vs-6-hour.matrix

<span class="co">#&#20063;&#21487;&#25351;&#23450;&#39034;&#24207;</span>
<span class="kw">rsem-generate-data-matrix</span> 1.fq.genes.results 3.fq.genes.results <span class="kw">&gt;</span> results.matrix</code></pre></div>
<h4 id="&#24046;&#24322;&#34920;&#36798;&#20998;&#26512;emseq">8. &#24046;&#24322;&#34920;&#36798;&#20998;&#26512;&#65288;EMSeq&#65289;</h4>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash"><span class="kw">gunzip</span> -c /data/nematostella.fa.gz <span class="kw">&gt;</span> ./nematostella.fa
<span class="kw">mkdir</span> ebseq
<span class="kw">cd</span> ebseq

<span class="kw">rsem-generate-data-matrix</span> /data/[0-9].fq.genes.results /data/10.fq.genes.results <span class="kw">&gt;</span> 0-vs-6-hour.matrix

<span class="kw">mkdir</span> ebseq
<span class="kw">cd</span> ebseq
<span class="kw">cp</span> ../rsem/0-vs-6-hour.matrix .

<span class="kw">rsem-run-ebseq</span> 0-vs-6-hour.matrix 5,5 0-vs-6-hour.changed
<span class="co">#.matrix file contains 2 conditions, each with 5 replicates; if you had two replicates, you would call rsem-run-ebseq with 2,2.</span>

<span class="co">#&#25552;&#21462;&#24046;&#24322;&#34920;&#36798;&#22522;&#22240;&#24182;&#32467;&#21512;&#20110;&#27880;&#37322;&#20449;&#24687;</span>
<span class="kw">python</span> /usr/local/share/eel-pond/extract-and-annotate-changed.py 0-vs-6-hour.changed /mnt/nematostella.fa 0-vs-6-hour.changed.csv

<span class="co">#&#21487;&#35270;&#21270;</span>
<span class="kw">python</span> /usr/local/share/eel-pond/plot-expression.py 0-vs-6-hour.matrix 5,5 0-vs-6-hour.changed.csv</code></pre></div>
<div class="figure">
<img src="http://khmer-protocols.readthedocs.org/en/ngs2014/_images/0-vs-6-hour.matrix.png" alt="" />

</div>
</body>
</html>
